src/main/java/uk/gov/hmcts/reform/prl/controllers/CallbackController.java

```diff
@@ -174,6 +182,26 @@ public AboutToStartOrSubmitCallbackResponse generateDocumentSubmitApplication(
         return AboutToStartOrSubmitCallbackResponse.builder().data(caseDataUpdated).build();
     }
 
+    @PostMapping(path = "/pre-populate-court-details", consumes = APPLICATION_JSON, produces = APPLICATION_JSON)
+    @ApiOperation(value = "Callback to pre  populate court details")
+    public AboutToStartOrSubmitCallbackResponse prePopulateCourtDetails(
+        @RequestBody uk.gov.hmcts.reform.ccd.client.model.CallbackRequest callbackRequest) throws NotFoundException {
+        CaseData caseData = CaseUtils.getCaseData(callbackRequest.getCaseDetails(), objectMapper);
+        Court closestChildArrangementsCourt = courtLocatorService
+            .getNearestFamilyCourt(caseData);
+        Optional<CourtEmailAddress> courtEmailAddress = courtLocatorService.getEmailAddress(closestChildArrangementsCourt);
+        Map<String, Object> caseDataUpdated = callbackRequest.getCaseDetails().getData();
+        if (courtEmailAddress.isPresent()) {
+            log.info("court email {} with case id {}", courtEmailAddress.get().getAddress(),caseData.getId());
```

Comment:
Lets not print court email here. Print generic message like court email is retrieved for case id etc.